internal/batch/batch_manager.go

```diff
@@ -588,3 +594,30 @@ func (bm *batchManager) LoadContexts(ctx context.Context, payload *DispatchPaylo
 	}
 	return nil
 }
+
+func (bm *batchManager) CancelBatch(ctx context.Context, batchID string) error {
+	id, err := fftypes.ParseUUID(ctx, batchID)
+	if err != nil {
+		return err
+	}
+	bp, err := bm.database.GetBatchByID(ctx, bm.namespace, id)
+	if err != nil {
+		return err
+	}
+	batch, err := bm.data.HydrateBatch(ctx, bp)
+	if err != nil {
+		return err
+	}
+	if len(batch.Payload.Messages) == 0 {
+		return i18n.NewError(ctx, coremsgs.MsgErrorLoadingBatch)
+	}
+	msg := batch.Payload.Messages[0]
+	processor, err := bm.getProcessor(msg.Header.TxType, msg.Header.Type, msg.Header.Group, msg.Header.SignerRef.Author, false)
+	if err != nil {
+		return err
+	}
+	if processor == nil {
+		return i18n.NewError(ctx, coremsgs.MsgCannotCancelBatchState)
```

Comment:
The message text is very clear, but the variable name `MsgCannotCancelBatchState` isn't - maybe instead:
`MsgBatchProcessorNotActive` ?